src/3d/qgspointcloudlayerchunkloader_p.cpp

```diff
@@ -112,7 +138,30 @@ void QgsPointCloud3DSymbolHandler::processNode( QgsPointCloudIndex *pc, const In
   attributes.push_back( QgsPointCloudAttribute( QStringLiteral( "X" ), QgsPointCloudAttribute::Int32 ) );
   attributes.push_back( QgsPointCloudAttribute( QStringLiteral( "Y" ), QgsPointCloudAttribute::Int32 ) );
   attributes.push_back( QgsPointCloudAttribute( QStringLiteral( "Z" ), QgsPointCloudAttribute::Int32 ) );
-  attributes.push_back( QgsPointCloudAttribute( QStringLiteral( "Classification" ), QgsPointCloudAttribute::Char ) );
+
+  std::unique_ptr< QgsPointCloudAttribute > parameterAttribute;
+  if ( mSymbol.get()->renderingStyle() == QgsPointCloud3DSymbol::ColorRamp )
```

Comment:
Move this logic to a virtual

    QgsPoiontCloud3DSymbol::usedAttributes() 

method, just like QgsPointCloudRenderer::usedAttributes